### ISSUE TITLE
Invalid value of type 'table' for setting 'double_buffer'. Expected value of type 'boolean'.

### ISSUE THREAD

**Issue by ghost** [brndnmtthws/conky#80]
$ conky -DD --double-buffer
DEBUG(0) [/tmp/makepkg/conky-git/src/conky/src/conky.cc:2671]: reading contents from config file '/home/emacs/.conkyrc'
conky: desktop window (1200003) is subwindow of root window (2b7)
conky: window type - desktop
conky: drawing to created window (0x2a00001)
conky: Invalid value of type 'table' for setting 'double_buffer'. Expected value of type 'boolean'.
DEBUG(1) [/tmp/makepkg/conky-git/src/conky/src/core.cc:1913]: no templates to replace
DEBUG(1) [/tmp/makepkg/conky-git/src/conky/src/core.cc:1913]: no templates to replace
DEBUG(1) [/tmp/makepkg/conky-git/src/conky/src/core.cc:576]: Adding $cpu for CPU 0
DEBUG(1) [/tmp/makepkg/conky-git/src/conky/src/core.cc:588]: Adding $cpubar for CPU 0
^Cconky: received SIGINT or SIGTERM to terminate. bye!

but this works:
$ conky -DD
DEBUG(0) [/tmp/makepkg/conky-git/src/conky/src/conky.cc:2671]: reading contents from config file '/home/emacs/.conkyrc'
conky: desktop window (1200003) is subwindow of root window (2b7)
conky: window type - desktop
conky: drawing to created window (0x1c00001)
conky: drawing to double buffer
DEBUG(1) [/tmp/makepkg/conky-git/src/conky/src/core.cc:1913]: no templates to replace
DEBUG(1) [/tmp/makepkg/conky-git/src/conky/src/core.cc:1913]: no templates to replace
DEBUG(1) [/tmp/makepkg/conky-git/src/conky/src/core.cc:576]: Adding $cpu for CPU 0
DEBUG(1) [/tmp/makepkg/conky-git/src/conky/src/core.cc:588]: Adding $cpubar for CPU 0

Using this config `~/.conkyrc`:

```
conky.config = {
    alignment = 'top_left',
    background = false,
    border_width = 1,
    cpu_avg_samples = 2,
  default_color = 'white',
    default_outline_color = 'white',
    default_shade_color = 'white',
    draw_borders = false,
    draw_graph_borders = true,
    draw_outline = false,
    draw_shades = false,
    use_xft = true,
    font = 'DejaVu Sans Mono:size=12',
    gap_x = 5,
    gap_y = 60,
    minimum_height = 5,
  minimum_width = 5,
    net_avg_samples = 2,
    no_buffers = true,
    double_buffer = true,
    out_to_console = false,
    out_to_stderr = false,
    extra_newline = false,
    own_window = true,
    own_window_class = 'Conky',
    own_window_type = 'desktop',
    stippled_borders = 0,
    update_interval = 1.0,
    uppercase = false,
    use_spacer = 'none',
    show_graph_scale = false,
    show_graph_range = false
}

conky.text = [[
${scroll 16 $nodename - $sysname $kernel on $machine | }
$hr
${color grey}Uptime:$color $uptime
${color grey}Frequency (in MHz):$color $freq
${color grey}Frequency (in GHz):$color $freq_g
${color grey}RAM Usage:$color $mem/$memmax - $memperc% ${membar 4}
${color grey}Swap Usage:$color $swap/$swapmax - $swapperc% ${swapbar 4}
${color grey}CPU Usage:$color $cpu% ${cpubar 4}
${color grey}Processes:$color $processes  ${color grey}Running:$color $running_processes
$hr
${color grey}File systems:
 / $color${fs_used /}/${fs_size /} ${fs_bar 6 /}
${color grey}Networking:
Up:$color ${upspeed eth0} ${color grey} - Down:$color ${downspeed eth0}
$hr
${color grey}Name              PID   CPU%   MEM%
${color lightgrey} ${top name 1} ${top pid 1} ${top cpu 1} ${top mem 1}
${color lightgrey} ${top name 2} ${top pid 2} ${top cpu 2} ${top mem 2}
${color lightgrey} ${top name 3} ${top pid 3} ${top cpu 3} ${top mem 3}
${color lightgrey} ${top name 4} ${top pid 4} ${top cpu 4} ${top mem 4}
]]
```

tested with commit c35c8520baa3ace11eefeb2aa9141901578022aa :

```
conky  -V
conky 2.0.0_pre compiled Wed Feb 18 00:54:36 CET 2015 for Linux 3.19.0-gbfa76d49 x86_64

Compiled in features:

System config file: /etc/conky/conky.conf
Package library path: /usr/lib/conky


 General:
  * math
  * hddtemp
  * portmon
  * Curl
  * RSS
  * Weather (METAR)
  * Weather (XOAP)
  * wireless
  * support for IBM/Lenovo notebooks
  * builtin default configuration
  * old configuration syntax
  * Imlib2
  * apcupsd
  * iostats
  * ncurses
  * Internationalization support
 X11:
  * Xdamage extension
  * Xft
  * ARGB visual
  * Own window

 Music detection:
  * MPD
  * MOC

 Default values:
  * Netdevice: eth0
  * Local configfile: $HOME/.conkyrc
  * Localedir: /usr/share/locale
  * Maximum netdevices: 64
  * Maximum text size: 16384
  * Size text buffer: 256

```

**Comment by emanueLczirai** [comment]
looks like this happens for others too `-d`
conky: Invalid value of type 'table' for setting 'background'. Expected value of type 'boolean'.

**Comment by emanueLczirai** [comment]
ok the value is `-2` instead of whatever values boolean should have (1 and 0 ?).
conky: Invalid value of type 'table' (-2) for setting 'double_buffer'. Expected value of type 'boolean'.

And this doesn't happen on setting (due to `-b`) but later on... I've yet to find out where

**Comment by emanueLczirai** [comment]
Stacktrace with correct line numbers (commit  c35c8520baa3ace11eefeb2aa9141901578022aa):

```
gdb -- conky
GNU gdb (GDB) 7.8.2
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-unknown-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from conky...done.
(gdb) run -DD -b
Starting program: /usr/bin/conky -DD -b
warning: Could not load shared library symbols for linux-vdso.so.1.
Do you need "set solib-search-path" or "set sysroot"?
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
DEBUG(0) [/tmp/makepkg/conky-git/src/conky/src/conky.cc:2671]: reading contents from config file '/home/emacs/.conkyrc'
conky: desktop window (1600003) is subwindow of root window (2b7)
conky: window type - desktop
conky: drawing to created window (0x3a00001)

Program received signal SIGTRAP, Trace/breakpoint trap.
conky::simple_config_setting<bool, conky::lua_traits<bool, true, false, false> >::do_convert (
    this=0x6dc300 <use_xdbe>, l=..., index=-2) at /tmp/makepkg/conky-git/src/conky/src/setting.hh:278
278				NORM_ERR("Invalid value of type '%s' for setting '%s'. "
(gdb) bt full
#0  conky::simple_config_setting<bool, conky::lua_traits<bool, true, false, false> >::do_convert (
    this=0x6dc300 <use_xdbe>, l=..., index=-2) at /tmp/makepkg/conky-git/src/conky/src/setting.hh:278
        index = -2
        l = @0x7054d0: {<std::mutex> = {<std::__mutex_base> = {_M_mutex = {__data = {__lock = 0, 
                  __count = 0, __owner = 0, __nusers = 0, __kind = 0, __spins = 0, __elision = 0, __list = {
                    __prev = 0x0, __next = 0x0}}, __size = '\000' <repeats 39 times>, 
                __align = 0}}, <No data fields>}, cobj = std::shared_ptr (count 1, weak 0) 0x706960}
        this = 0x6dc300 <use_xdbe>
#1  0x0000000000430f2d in conky::simple_config_setting<bool, conky::lua_traits<bool, true, false, false> >::lua_setter (this=0x6dc300 <use_xdbe>, l=..., init=<optimized out>)
    at /tmp/makepkg/conky-git/src/conky/src/setting.hh:298
        s = {L = 0x7054d0, n = 5}
        ok = true
#2  0x000000000047fc0d in priv::use_xdbe_setting::lua_setter (this=0x6dc300 <use_xdbe>, l=..., 
    init=<optimized out>) at /tmp/makepkg/conky-git/src/conky/src/x11.cc:163
        s = {L = 0x7054d0, n = 5}
#3  0x000000000045ea37 in conky::priv::config_setting_base::process_setting (l=..., init=init@entry=true)
    at /tmp/makepkg/conky-git/src/conky/src/setting.cc:137
        s = {L = 0x7054d0, n = 4}
        ptr = <optimized out>
#4  0x000000000045f12d in conky::set_config_settings (l=...)
    at /tmp/makepkg/conky-git/src/conky/src/setting.cc:218
        i = 100
        s = {L = 0x7054d0, n = <optimized out>}
#5  0x000000000043559c in initialisation (argc=argc@entry=3, argv=argv@entry=0x7fffffffe028)
    at /tmp/makepkg/conky-git/src/conky/src/conky.cc:2970
        act = {__sigaction_handler = {sa_handler = 0x709510, sa_sigaction = 0x709510}, sa_mask = {__val = {
              7332544, 0, 140737351974416, 140737310084976, 0, 0, 140737310085771, 140733193388032, 7381168, 
              7381128, 7041872, 0, 7361744, 0, 0, 0}}, sa_flags = -136407603, sa_restorer = 0x1}
        oact = {__sigaction_handler = {sa_handler = 0x0, sa_sigaction = 0x0}, sa_mask = {__val = {
              140737349681480, 140737488347176, 3, 0, 140737351974416, 4785512, 140737349737070, 7361744, 
              140737330758057, 7361744, 16, 140737488346720, 140737488346640, 0, 140737310562631, 1}}, 
          sa_flags = 3, sa_restorer = 0x3}
#6  0x00000000004294bf in main (argc=3, argv=0x7fffffffe028)
    at /tmp/makepkg/conky-git/src/conky/src/conky.cc:3135
No locals.
(gdb) 

```

```diff
diff --git a/src/setting.hh b/src/setting.hh
index e7b2064..5cf6901 100644
--- a/src/setting.hh
+++ b/src/setting.hh
@@ -274,6 +274,7 @@ namespace conky {
                        return {default_value, true};
 
                if(l.type(index) != Traits::type) {
+                       asm("int $3");
                        NORM_ERR("Invalid value of type '%s' for setting '%s'. "
                                         "Expected value of type '%s'.", l.type_
name(l.type(index)),
                                         Base::name.c_str(), l.type_name(Traits:
:type) );

```

**Comment by marcpayne** [comment]
Issue #97 is a duplicate of this issue. By any chance, could you apply the patch in commit 11ec0105612 and see if it solves the problem for you?

**Comment by marcpayne** [comment]
Hmm, since @ghost is a ghost now, I'll just ping this issue... Fixed by #119.

**Comment by lasers** [comment]
Ping received. Allegedly fixed by #119.
